src/metrics/metrics.go

```diff
@@ -87,7 +87,13 @@ var (
 	)
 )
 
-func init() {
+func Init(prometheusExporterMetricsDisabled bool) {
+	if prometheusExporterMetricsDisabled {
```

Comment:
does this work b/c it wipes out a bunch of already registered metrics for the go runtime? i suppose that's the best way to do this?

Reply:
Yup, the `prometheus` package registers `NewProcessCollector` and `NewGoCollector` to the internal registry in `init()` and this replaces them with new empty ones before registering metrics.

I'm not clear on what's the best way to do this, I just took inspiration from other exporters.

Now that you ask through, I see `Unregister()` on the `Registry` interface. I'm going to try that to get the same behavior, if that works that means we can keep the `init()` function and just unregister these two default collectors after parsing the flags.

Reply:
Don't want to leave this comment hanging. I did look into `Unregister()` some, but ran into problems with the fact that I needed to know exactly what to unregister.. and there is no way of having that information given how the `Registry` creates them in its own `init()` function and the interface doesn't expose a way to peek into what `Collector`(s) are registered.